### ISSUE TITLE
AppRailTitle fires off on:click function twice

### ISSUE THREAD

**Issue by piperun** [skeletonlabs/skeleton#1195]
### Current Behavior

At the moment when you got `value` & `selected` but also `on:click`, it'll fire off the `on:click` function twice

### Steps To Reproduce

1. Essentially create an `AppRailTitle` & `AppRail`
2. add the respective `selected` store to `AppRail` and `value` to `AppRailTitle`s 
3. add `on:click` to `AppRailTitle`
4. click and it will run the click function twice

### Anything else?

Example code:
```js
    const menuStore = writable(false);
    const storeVar = writable(1);
    let calls: number = 0;
    let prevApp: number;
    const menuOpen = () => {
        console.log(calls++) // this will count twice instead of just once when clicked
        if ($menuStore && $storeValue == prevApp) {
            $menuStore = false;
        } else {
            prevApp = $storeValue;
            $menuStore = true;
        }
    };


    <AppRail selected={storeVar}>
        <AppRailTile label="Create" title="Create" value={1} on:click={menuOpen}>
              <!-- slot -->
        </AppRailTile>
        <AppRailTile label="View" title="View" value={2} on:click={menuOpen}>
              <!-- slot -->
        </AppRailTile>
    </AppRail>
              <!-- logic & element for menu -->
</div>
```

discord convo:
https://ptb.discord.com/channels/1003691521280856084/1088561593534660648

**Comment by endigo9740** [comment]
Thanks @piperun I'l aim to review tomorrow and get you answer on what's up here

**Comment by endigo9740** [comment]
@piperun I just rolled out a quick fix for this on dev branch. I typically handle this through a PR, but given the simplicity of the change I cheated a bit and did a direct commit:
https://github.com/skeletonlabs/skeleton/commit/279122e4a8404573e2fb250de193906958f2cbae

In short, what's happening here is we typically only dispatch CUSTOM events. In this we we opted to dispatch `click` (a standard event), while also putting in the `on:click` forward clicks for custom use as well. So we ended up doubling the click event this way. Dropping the additional `on:click` resolved this though.
